lib/rules/no-container.ts

```diff
@@ -120,6 +99,47 @@ export default ESLintUtils.RuleCreator(getDocsUrl)<Options, MessageIds>({
             });
         }
       },
+
+      VariableDeclarator(node) {
+        const initIdentifierNode = getDeepestIdentifierNode(node.init);
+
+        const isRenderWrapperVariableDeclarator = initIdentifierNode
+          ? renderWrapperNames.includes(initIdentifierNode.name)
+          : false;
+
+        if (
+          !helpers.isRenderVariableDeclarator(node) &&
+          !isRenderWrapperVariableDeclarator
+        ) {
+          return;
+        }
```

Comment:
The rule implementation is mostly the same, except this bit to detect _reportable_ `render` or some wrapper around it.